File: climb-log-flutter/lib/climasys/widgets/upsert_route_dialog/models/edit_route_command.dart
class EditRouteCommand {
  final int routeId;
  final int gradeId;
  final int? standardGradeId;
  final int? competitionId;
  final int? pointValue;
  final int? sectorId;

  EditRouteCommand({
    required this.routeId,
    required this.gradeId,
    this.standardGradeId,
    this.competitionId,
    this.pointValue,
    this.sectorId
  });

  Map<String, dynamic> toJson() =>
    {
      'routeId': routeId,
      'gradeId': gradeId,
      'standardGradeId': standardGradeId,
      'competitionId': competitionId,
      'pointValue': pointValue,
      'sectorId': sectorId
    };
}

File: ClimbLog.Api/ClimbLog.Application/Features/Routes/Commands/EditRoute/EditRouteCommand.cs
using MediatR;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClimbLog.Application.Features.Routes.Commands.EditRoute;
public class EditRouteCommand : IRequest<Unit>
{
    public int RouteId { get; set; }
    public int GradeId { get; set; }
    public int? StandardGradeId { get; set; }
    public int? CompetitionId { get; set; }
    public int? PointValue { get; set; }
    public int? SectorId { get; set; }
}
